src/main/scala/is/hail/driver/MinRep.scala

```diff
@@ -0,0 +1,70 @@
+package is.hail.driver
+
+import is.hail.utils._
+import is.hail.variant._
+import org.kohsuke.args4j.{Option => Args4jOption}
+
+object MinRep extends Command {
+
+  class Options extends BaseOptions {
+  }
+
+  def newOptions = new Options
+
+  def name = "minrep"
+
+  def description = "Gives minimal left-aligned representation of alleles in the VDS"
+
+  def supportsMultiallelic = true
+
+  def requiresVDS = true
+
+  def splitGT(gt: Int, i: Int): Int = {
+    val p = Genotype.gtPair(gt)
+    (if (p.j == i) 1 else 0) +
+      (if (p.k == i) 1 else 0)
+  }
+
+  def minRep(start: Int, ref: String, alt: Array[AltAllele]): (Int, String, Array[String]) = {
+    val alts = alt.map(a => a.alt)
+    require(alts.forall(ref != _))
+
+    val min_length = math.min(ref.length, alts.map(x => x.length).min)
+    var ne = 0
+
+    while( ne < min_length - 1
+      && alts.forall(x => ref(ref.length - ne - 1) == x(x.length - ne - 1))
+    ){
+      ne += 1
+    }
+
+    var ns = 0
+    while(ns < min_length - ne -1
+      && alts.forall(x => ref(ns) == x(ns))
+    ){
+      ns += 1
+    }
+
+    assert(ns < ref.length - ne && alts.forall(x => ns < x.length - ne))
+    (start + ns, ref.substring(ns, ref.length - ne), alts.map(x => x.substring(ns, x.length - ne)))
+  }
+
+
+  def run(state: State, options: Options): State = {
+    val vds = state.vds
+    if (vds.wasSplit) {
+      warn("called redundant `splitmulti' on an already split VDS")
```

Comment:
Also, is this necessarily bad? I'm about to run minrep on an already split VDS (with some un-minrepped variants after `filter_alleles` - though I suppose this will change eventually and be moot).